pkg/utils/gpu.go

```diff
@@ -17,124 +17,81 @@ limitations under the License.
+
+	NvidiaGridDriverVersion = "535.161.08"
+	AKSGPUGridVersionSuffix = "20241021235607"
 )
 
-func GetAKSGPUImageSHA(size string) string {
-	if UseGridDrivers(size) {
-		return AKSGPUGridSHA
-	}
-	return AKSGPUCudaSHA
+type NvidiaSKUConfig struct {
+	NvidiaEnabledSKUFamilies        map[string][]string `yaml:"nvidiaEnabledSKUs"`
+	MarinerNvidiaEnabledSKUFamilies map[string][]string `yaml:"marinerNvidiaEnabledSKUs"`
 }
 
 var (
-	/* If a new GPU sku becomes available, add a key to this map, but only if you have a confirmation
-	   that we have an agreement with NVIDIA for this specific gpu.
-	*/
-	NvidiaEnabledSKUs = map[string]bool{
-		// M60
-		"standard_nv6":      true,
-		"standard_nv12":     true,
-		"standard_nv12s_v3": true,
-		"standard_nv24":     true,
-		"standard_nv24s_v3": true,
-		"standard_nv24r":    true,
-		"standard_nv48s_v3": true,
-		// P40
-		"standard_nd6s":   true,
-		"standard_nd12s":  true,
-		"standard_nd24s":  true,
-		"standard_nd24rs": true,
-		// P100
-		"standard_nc6s_v2":   true,
-		"standard_nc12s_v2":  true,
-		"standard_nc24s_v2":  true,
-		"standard_nc24rs_v2": true,
-		// V100
-		"standard_nc6s_v3":   true,
-		"standard_nc12s_v3":  true,
-		"standard_nc24s_v3":  true,
-		"standard_nc24rs_v3": true,
-		"standard_nd40s_v3":  true,
-		"standard_nd40rs_v2": true,
-		// T4
-		"standard_nc4as_t4_v3":  true,
-		"standard_nc8as_t4_v3":  true,
-		"standard_nc16as_t4_v3": true,
-		"standard_nc64as_t4_v3": true,
-		// A100 40GB
-		"standard_nd96asr_v4":       true,
-		"standard_nd112asr_a100_v4": true,
-		"standard_nd120asr_a100_v4": true,
-		// A100 80GB
-		"standard_nd96amsr_a100_v4":  true,
-		"standard_nd112amsr_a100_v4": true,
-		"standard_nd120amsr_a100_v4": true,
-		// A100 PCIE 80GB
-		"standard_nc24ads_a100_v4": true,
-		"standard_nc48ads_a100_v4": true,
-		"standard_nc96ads_a100_v4": true,
-		"standard_ncads_a100_v4":   true,
-		// A10
-		"standard_nc8ads_a10_v4":  true,
-		"standard_nc16ads_a10_v4": true,
-		"standard_nc32ads_a10_v4": true,
-		// A10, GRID only
-		"standard_nv6ads_a10_v5":   true,
-		"standard_nv12ads_a10_v5":  true,
-		"standard_nv18ads_a10_v5":  true,
-		"standard_nv36ads_a10_v5":  true,
-		"standard_nv36adms_a10_v5": true,
-		"standard_nv72ads_a10_v5":  true,
-		// A100
-		"standard_nd96ams_v4":      true,
-		"standard_nd96ams_a100_v4": true,
+	nvidiaEnabledSKUs        = make(map[string]bool)
+	marinerNvidiaEnabledSKUs = make(map[string]bool)
+)
+
+//go:embed supported-gpus.yaml
+var configFile []byte
+
+func init() {
+	readNvidiaSKUConfig()
+}
+
+func readNvidiaSKUConfig() {
+	var nvidiaSKUConfig NvidiaSKUConfig
+
+	err := yaml.Unmarshal(configFile, &nvidiaSKUConfig)
+	if err != nil {
+		panic(err)
 	}
+	for _, skus := range nvidiaSKUConfig.NvidiaEnabledSKUFamilies {
+		for _, sku := range skus {
+			nvidiaEnabledSKUs[sku] = true
+		}
+	}
+	for _, skus := range nvidiaSKUConfig.MarinerNvidiaEnabledSKUFamilies {
+		for _, sku := range skus {
+			marinerNvidiaEnabledSKUs[sku] = true
+		}
+	}
+}
```

Comment:
why not just have the type we marshal into be our actual structure rather than iterating through and creating a duplicate structure here? Its less than 100 items so not an expensive operation just wondering if there is a need to iterate through the list at all? Can we not just use the nvidiaSKUConfig type directly?

Reply:
I presume it is because the layout convenient for authoring and maintenance is different from what is required at runtime for lookup. But, whatever the case, the list (and the code) is essentially copied from elsewhere that is considered the master source in this context, the main value for this codebase to close alignment to its upstream.